utilities/test_suite/rpp_test_suite_common.h

```diff
@@ -394,20 +577,105 @@ inline void convert_pkd3_to_pln3(Rpp8u *input, RpptDescPtr descPtr)
     free(inputCopy);
 }
 
-inline void read_image_batch_opencv(Rpp8u *input, RpptDescPtr descPtr, string imageNames[])
+// Opens a folder and recursively search for .jpg files
+void open_folder(const string& folderPath, vector<string>& imageNames, vector<string>& imageNamesPath)
+{
+    auto src_dir = opendir (folderPath.c_str());
+    struct dirent* entity;
+    std::string fileName = " ";
+
+    if (src_dir == nullptr)
+        std::cerr << "\n ERROR: Failed opening the directory at " <<folderPath;
+
+    while((entity = readdir (src_dir)) != nullptr)
+    {
+        string entry_name(entity->d_name);
+        if (entry_name == "." || entry_name == "..")
+            continue;
+        fileName = entity->d_name;
+        std::string filePath = folderPath;
+        filePath.append("/");
+        filePath.append(entity->d_name);
+        fs::path pathObj(filePath);
+        if(fs::exists(pathObj) && fs::is_directory(pathObj))
+        {
+            open_folder(filePath, imageNames, imageNamesPath);
+        }
+        if (fileName.size() > 4 && fileName.substr(fileName.size() - 4) == ".jpg")
+        {
+            imageNamesPath.push_back(filePath);
+            imageNames.push_back(entity->d_name);
+        }
+    }
+    if(imageNames.empty())
+        std::cerr << "\n Did not load any file from " << folderPath;
+
+    closedir(src_dir);
+}
+
+// Searches for .jpg files in input folders
+void search_jpg_files(const string& folder_path, vector<string>& imageNames, vector<string>& imageNamesPath)
+{
+    vector<string> entry_list;
+    string full_path = folder_path;
+    auto sub_dir = opendir(folder_path.c_str());
+    if (!sub_dir)
+    {
+        std::cerr << "ERROR: Failed opening the directory at "<< folder_path << std::endl;
+        exit(0);
+    }
+
+    struct dirent* entity;
+    while ((entity = readdir(sub_dir)) != nullptr)
+    {
+        string entry_name(entity->d_name);
+        if (entry_name == "." || entry_name == "..")
+            continue;
+        entry_list.push_back(entry_name);
+    }
+    closedir(sub_dir);
+    std::sort(entry_list.begin(), entry_list.end());
```

Comment:
why do we need to sort the entry_list?

Reply:
Removed sorting entry list here as we are doing sorting in cpp files